### PR TITLE
drivers: wifi: Update RPU patch for RevB to #48

### PR DESCRIPTION
**Pull request by karun2796**
UMAC:
 * Add support for HE TB PPDU in radio test mode
 * Add support for MAC address change from host
 * Populating RF parameters from OTP
 * Add delay after sending BA event to host

LMAC:
 * Fix for performance interoperability issues with some APs

PHY:
 * Alternative mechanism to calibrate XO by measuring CW freq offset

Also, update the interface header files.

Signed-off-by: Karun Kumar Eagalapati <karun.kumar@nordicsemi.no>

### CODE REVIEW

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h [hunk 1]
```diff
@@ -1391,6 +1392,23 @@ struct nrf_wifi_umac_cmd_mcast_filter {
 } __NRF_WIFI_PKD;
 
 
+/**
+ * struct nrf_wifi_umac_cmd_change_macaddr - Set Mac Address.
```
Comment (krish2718) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h:1396]:
```suggestion
 * struct nrf_wifi_umac_cmd_change_macaddr - Change Mac Address.
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h [hunk 2]
```diff
@@ -1391,6 +1392,23 @@ struct nrf_wifi_umac_cmd_mcast_filter {
...
+ *
+ *    - This has to be used when interface is down.
+ *
+ * @umac_hdr: Refer to &struct img_umac_hdr.
+ * @mac_addr : Mac_addr to be programmed
```
Comment (krish2718) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h:1401]:
```suggestion
 * @mac_addr : New MAC address
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h [hunk 3]
```diff
@@ -1391,6 +1392,23 @@ struct nrf_wifi_umac_cmd_mcast_filter {
...
+ *
+ */
+struct nrf_wifi_umac_change_macaddr_info {
```
Comment (krish2718) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h:1404]: we can drop the `_info`

**Review by krish2718 — CHANGES_REQUESTED** [review]
* Divide in to two commits with separate patch (1 logical change per-commit)
* Describe the `changes`
* It would be good to have a driver commit that uses these functions, else there would be no users.
nit
* Commit title should not end with full stop

**Review by sr1dh48r — APPROVED** [review]

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h (hunk 2 above)
Comment (karun2796) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h]: Suggested change was done.

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h (hunk 3 above)
Comment (karun2796) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw_B/host_rpu_umac_if.h:1404]: As discussed, This we are keeping it as it is.

**Review by sachinthegreen — APPROVED** [review]

**Review by krish2718 — APPROVED** [review]
As discussed offline the two comments about separate commit for headers and having driver changes are not necessary.

**Review by bama-nordic — APPROVED** [review]

**Pull request merged**